webapp/src/selectors.js

```diff
@@ -1,9 +1,82 @@
+import {createSelector} from 'reselect';
+
+import {Preferences} from 'mattermost-redux/constants';
+import {get} from 'mattermost-redux/selectors/entities/preferences';
+import {getMyTeams} from 'mattermost-redux/selectors/entities/teams';
+
 import {id as pluginId} from './manifest';
 
+const MESSAGE_DISPLAY = 'message_display';
+const MESSAGE_DISPLAY_DEFAULT = 'clean';
+const MESSAGE_DISPLAY_COMPACT = 'compact';
+
 function getPluginState(state) {
     return state['plugins-' + pluginId] || {};
 }
 
 export function getConfirmationModalState(state) {
     return getPluginState(state).confirmationModal || {};
-}
\ No newline at end of file
+}
+
+export function getWindowWidth(state) {
+    return getPluginState(state).windowWidth;
+}
+
+export function isSurveyPostSmall(state, isRHS) {
+    return isRHS || !useFullSurveyPost(state);
+}
+
+function isCompactView(state) {
+    const compactView = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, MESSAGE_DISPLAY, MESSAGE_DISPLAY_DEFAULT) === MESSAGE_DISPLAY_COMPACT;
```

Comment:
Could you maybe just return the result of this line and not worry about setting a variable?

Reply:
Good catch. I think I must've been logging this value at some point